docker/build_artifacts/sagemaker/serve.py

```diff
@@ -89,30 +92,33 @@ def __init__(self):
             parts = self._sagemaker_port_range.split("-")
             low = int(parts[0])
             hi = int(parts[1])
-            self._tfs_grpc_port = []
-            self._tfs_rest_port = []
             if low + 2 * self._tfs_instance_count > hi:
                 raise ValueError("not enough ports available in SAGEMAKER_SAFE_PORT_RANGE ({})"
                                  .format(self._sagemaker_port_range))
-            self._tfs_grpc_port_range = "{}-{}".format(low,
-                                                       low + 2 * self._tfs_instance_count)
-            self._tfs_rest_port_range = "{}-{}".format(low + 1,
-                                                       low + 2 * self._tfs_instance_count + 1)
-            for i in range(self._tfs_instance_count):
-                self._tfs_grpc_port.append(str(low + 2 * i))
-                self._tfs_rest_port.append(str(low + 2 * i + 1))
-            # set environment variable for python service
-            os.environ["TFS_GRPC_PORT_RANGE"] = self._tfs_grpc_port_range
-            os.environ["TFS_REST_PORT_RANGE"] = self._tfs_rest_port_range
+            # split ports range here for grpc and rest port
+            rest_port = low
+            grpc_port = (low + hi) // 2
```

Comment:
Please note that if ```hi == low + 1```, then ```(low + hi) // 2 == low``` which is not what we want, sure with the current proposal we would never get into that problem since there's a check ```if low + 2 * self._tfs_instance_count > hi:``` but as explained earlier, that check expects the difference between hi and low to be at least 2 (which may be unintuitive).